docs: testing graphQL servers (#4374)
graphql/graphql-js: website/pages/docs/testing-approaches.mdx

- Receives input arguments and context as expected
- Calls the correct business logic
- Handles errors properly
- Returns the correct result

### When to use unit tests

Unit tests are fast and provide tight feedback loops. They're especially useful when:

- Developing new resolvers
- Validating error handling and edge cases
- Refactoring resolver logic and needing immediate verification

However, unit tests have limitations. Because you're mocking inputs and skipping the schema 
entirely, you won’t catch issues with how your schema connects to your resolver functions. 
There's also a risk of false positives if your mocks drift from real usage over time.

### Example: Unit test for a resolver

This test verifies that the resolver produces the expected result using mocked inputs.

```javascript
const result = await myResolver(parent, args, context);
expect(result).toEqual(expectedResult);
```

## Integration tests

Integration tests go a step further by testing resolvers and the schema together. 
You can run actual GraphQL queries and verify the end-to-end behavior within your
application's boundaries.

You can use the `graphql()` function from the GraphQL package, no HTTP server 

> needed. With the `graphql()` function, you can test the full flow: query > schema > 
> resolver >  data source (mocked or real).

Reviewer: ```suggestion
needed. With the `graphql()` function, you can test the full flow: request &gt; schema &gt; 
resolver &gt; data source (mocked or real).
```

Integration tests confirm that:

- The schema is correctly wired to resolvers
- Resolvers behave as expected when called through a query
- Data sources return expected results

### When to use integration tests

Use integration tests when:

> - You want to test the full operation flow from query to result

Reviewer: ```suggestion
- You want to test the full operation flow from request to result
```

> - You're testing how resolvers handle variables, fragments, and nested queries

Reviewer: ```suggestion
- You're testing how resolvers handle variables, fragments, and nested fields
```
